.github/workflows/_cleanup.yml

```diff
@@ -40,6 +40,9 @@ jobs:
                   elif [[ $clusterstate == *"Stopped"* ]] || [[ $clusterstate == *"Failed"* ]]; then
                       az aks delete --resource-group $RESOURCE_GROUP --name $name --no-wait --yes
                       break  # Exit the loop on successful attempt
+                  elif [[ $clusterstate == *"Deleting"* ]]; then
+                      echo "Returned cluster provisioning state is $clusterstate, continuing..."
+                      break  # Exit the loop on successful attempt
```

Comment:
is there a chance that the deletion process fails even if the cluster's state indicates `deleting`? in that scenario, the cluster would remain active while the cleanup job would report success?

Reply:
I honestly am not sure, but I know that in my pipeline that failed, it's status was reported as deleting and it went back to the loop to retry and it failed because the cluster didn't exist. So we could add a check to see if the cluster exists before even attempting to delete instead of doing this.

Reply:
If deletion process fails if the cluster state indicates deleting, that's an AKS backend issue. I don't think we should consider scenarios like this, given that this is not customer facing and this won't cause any outage or something.

Reply:
I think my concern was more about cost. If a cluster is created but not deleted and we don't catch it for a while, then it would have cost us a lot.